### PR TITLE
Add string interpolation to Console.WriteLine

### LINKED ISSUE #380: Add string interpolation to Console.WriteLine
We already suggest a code fix for `string.Format` (on `CC0048`) to use string interpolation. Let's suggest one for `Console.WriteLine`.

Should work with:

* `WriteLine(String, Object)`
* `WriteLine(String, Object[])`
* `WriteLine(String, Object, Object)`
* `WriteLine(String, Object, Object, Object)`
* `WriteLine(String, Object, Object, Object, Object)`

This:

````csharp
Console.WriteLine("some {0} text", adjective);
````

Becomes:

````csharp
Console.WriteLine($"some {adjective} text");
````

Category: `Style`
Severity: `Info`
Diagnostic id: `CC0095`

More info on `Console.WriteLine` on [MSDN](https://msdn.microsoft.com/en-us/library/828t9b9h%28v=vs.110%29.aspx).

### PR DESCRIPTION
**Pull request by giggio**
Closes #380
Already using localization
Updated String.Format analyzer and code fix to use localization
and to the more performant code action.
Using common code for string.format and console.writeline.

### CODE REVIEW

File: src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs [hunk 1]
```diff
@@ -28,21 +29,24 @@ public class StringFormatAnalyzer : DiagnosticAnalyzer
...
+            if (memberExpresion?.Name?.ToString() != methodName) return;
             var memberSymbol = context.SemanticModel.GetSymbolInfo(memberExpresion).Symbol;
             if (memberSymbol == null) return;
-            if (!memberSymbol.ToString().StartsWith("string.Format(string, ")) return;
+            if (!memberSymbol.ToString().StartsWith(methodOverloadSignature)) return;
             var argumentList = invocationExpression.ArgumentList as ArgumentListSyntax;
             if (argumentList?.Arguments.Count < 2) return;
             if (!argumentList.Arguments[0]?.Expression?.IsKind(SyntaxKind.StringLiteralExpression) ?? false) return;
-            if (memberSymbol.ToString() == "string.Format(string, params object[])" && argumentList.Arguments.Skip(1).Any(a => context.SemanticModel.GetTypeInfo(a.Expression).Type.TypeKind == TypeKind.Array)) return;
+            if (memberSymbol.ToString() == methodWithArraySignature && argumentList.Arguments.Skip(1).Any(a => context.SemanticModel.GetTypeInfo(a.Expression).Type.TypeKind == TypeKind.Array)) return;
```
Comment (baks) [inline src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs]: `argumentList.Arguments.Skip(1).Any(a => context.SemanticModel.GetTypeInfo(a.Expression).Type.TypeKind == TypeKind.Array)` can be computed once

File: src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs [hunk 2]
```diff
@@ -20,16 +21,24 @@ public class StringFormatCodeFixProvider : CodeFixProvider
...
 
-        private async Task<Document> MakeStringInterpolationAsync(Document document, InvocationExpressionSyntax invocationExpression, CancellationToken cancellationToken)
+        private async Task<Document> MakeStringInterpolationAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var invocationExpression = root.FindToken(diagnostic.Location.SourceSpan.Start).Parent.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First();
+            var newStringInterpolation = await CreateNewStringInterpolationAsync(document, invocationExpression, cancellationToken);
+            var newRoot = root.ReplaceNode(invocationExpression, newStringInterpolation);
+            var newDocument = document.WithSyntaxRoot(newRoot);
+            return newDocument;
```
Comment (baks) [inline src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs]: It's nothing important, but there is no need to creating variable `newDocument`, it just can return `document.WithSyntaxRoot(newRoot);`

File: src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs (hunk 1 above)
Comment (giggio) [inline src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs]: @baks what do you mean? It is being computed once.

File: src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs (hunk 2 above)
Comment (giggio) [inline src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs]: @baks The compiler optimizes these details away.

File: src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs (hunk 1 above)
Comment (baks) [inline src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs]: @giggio, my mistake

File: src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs (hunk 2 above)
Comment (baks) [inline src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs]: @giggio Yes, but I was concerning reading the code.

**Pull request merged**